src/main/java/nl/knmi/geoweb/backend/product/sigmet/Sigmet.java

```diff
@@ -50,917 +52,991 @@
+			return "Missing geometry";
+		}
+		if (!((Feature)effectiveStartGeometry).getProperty("selectionType").equals("box")&&
+				!((Feature)effectiveStartGeometry).getProperty("selectionType").equals("fir")&&
+				!((Feature)effectiveStartGeometry).getProperty("selectionType").equals("point"))	{
+			GeoJsonObject intersected=this.extractSingleStartGeometry();
+			int sz=((Polygon)((Feature)intersected).getGeometry()).getCoordinates().get(0).size();
+			if (sz<=7)  {
+				effectiveStartGeometry = intersected; // Use intersection result
+			}
+		}
+		StringBuilder sb = new StringBuilder();
+		String validdateFormatted = String.format("%02d", this.validdate.getDayOfMonth()) + String.format("%02d", this.validdate.getHour()) + String.format("%02d", this.validdate.getMinute());
+		String validdateEndFormatted = String.format("%02d", this.validdate_end.getDayOfMonth()) + String.format("%02d", this.validdate_end.getHour()) + String.format("%02d", this.validdate_end.getMinute());
+
+		sb.append(this.location_indicator_icao).append(" SIGMET ").append(this.sequence).append(" VALID ").append(validdateFormatted).append('/').append(validdateEndFormatted).append(' ').append(this.location_indicator_mwo).append('-');
+		sb.append('\n');
+		sb.append(this.location_indicator_icao).append(' ').append(this.firname);
+		if (this.cancels != null && this.cancelsStart != null) {
+			String validdateCancelled = String.format("%02d", this.cancelsStart.getDayOfMonth()) + String.format("%02d", this.cancelsStart.getHour()) + String.format("%02d", this.cancelsStart.getMinute());
+
+			sb.append(' ').append("CNL SIGMET ").append(this.cancels).append(" ").append(validdateCancelled).append('/').append(validdateEndFormatted);
+			return sb.toString();
+		}
+		sb.append('\n');
+		
+
+		if (va_extra_fields != null) {
+			sb.append(va_extra_fields.toTAC());
+		}
+		
+		sb.append(this.phenomenon.getShortDescription());
+
+
+		sb.append('\n');
+		if (this.getObs_or_forecast()!=null) {
+			sb.append(this.obs_or_forecast.toTAC());
+			sb.append('\n');
+		}
+		sb.append(this.featureToTAC((Feature)effectiveStartGeometry, FIR));
+		sb.append('\n');
+
+		String levelInfoText=this.levelinfo.toTAC();
+		if (!levelInfoText.isEmpty()) {
+			sb.append(levelInfoText);
+			sb.append('\n');
+		}
+
+		if (this.movement_type==null) {
+			this.movement_type=SigmetMovementType.STATIONARY;
+		}
+
+		switch (this.movement_type) {
+		case STATIONARY:
+			sb.append("STNR ");
+			break;
+		case MOVEMENT:
+			if (this.movement!=null) {
+				sb.append(this.movement.toTAC());
+				sb.append('\n');
+			}
+			break;
+		case FORECAST_POSITION:
+			// Present forecast_position geometry below
+			break;
+		}
+
+		if (this.change!=null) {
+			sb.append(this.change.toTAC());
+			sb.append('\n');
+		}
+		// TODO: Check why endGeometry is not used
+		GeoJsonObject endGeometry=this.findEndGeometry(((Feature)effectiveStartGeometry).getId());
```

Comment:
This endGeometry using findEndGeometry(((Feature)effectiveStartGeometry).getId()) is a leftover; now using findEndGeometry(((Feature)findStartGeometry()).getId()), FIR) fore FORECAST_POSITION. This line can be removed